packages/salesforcedx-vscode-core/test/vscode-integration/providers/configAggregatorProvider.test.ts

```diff
@@ -0,0 +1,205 @@
+      // Arrange
+      getCurrentDirectoryStub
+        .onCall(0)
+        .returns(ConfigAggregatorProvider.defaultBaseProcessDirectoryInVSCE);
+
+      // Act
+      const configAggregator = await (configAggregatorProvider as any).createConfigAggregator();
+
+      // Assert
+      // createConfigAggregator should store the cwd initially,
+      // and check it again after creating the ConfigAggregator
+      // to ensure that the cwd is set back to its original value.
+      expect(getCurrentDirectoryStub.callCount).to.equal(2);
+      // createConfigAggregator should change the process to the
+      // root project workspace path to create the ConfigAggregator,
+      // then change back to the original directory.
+      expect(changeCurrentDirectoryToStub.callCount).to.equal(2);
+      expect(changeCurrentDirectoryToStub.getCall(1).args[0]).to.equal(
+        ConfigAggregatorProvider.defaultBaseProcessDirectoryInVSCE
+      );
+      expect(configAggregatorCreateSpy.callCount).to.equal(1);
+      expect(configAggregator).to.not.equal(undefined);
+    });
+
+    it('should create a ConfigAggregator when inside of a project directory', async () => {
+      // Arrange
+      getCurrentDirectoryStub.returns(dummyProjectRootWorkspacePath);
+      getRootWorkspacePathStub.returns(dummyProjectRootWorkspacePath);
+
+      // Act
+      const configAggregator = await (configAggregatorProvider as any).createConfigAggregator();
+
+      // Assert
+      // createConfigAggregator should store the cwd initially,
+      // and check it again after creating the ConfigAggregator
+      // to ensure that the cwd is set back to its original value.
+      expect(getCurrentDirectoryStub.callCount).to.equal(2);
+      // Since the stubbed current directory is the same as the root
+      // workspace path, the directory should not have been changed.
+      expect(changeCurrentDirectoryToStub.callCount).to.equal(0);
+      expect(configAggregatorCreateSpy.callCount).to.equal(1);
+      expect(configAggregator).to.not.equal(undefined);
+      expect(getRootWorkspacePathStub.callCount).to.equal(1);
+    });
+
+    it('should create an SfdxConfigAggregator', async () => {
+      // Arrange
+      getCurrentDirectoryStub
+        .onCall(0)
+        .returns(ConfigAggregatorProvider.defaultBaseProcessDirectoryInVSCE);
+      const sfdxConfigAggregatorCreateSpy = sandbox.spy(
+        SfdxConfigAggregator,
+        'create'
+      );
+
+      // Act
+      const sfdxConfigAggregator = await (configAggregatorProvider as any).createConfigAggregator(
+        { sfdx: true }
+      );
+
+      // Assert
+      expect(sfdxConfigAggregatorCreateSpy.callCount).to.equal(1);
+      expect(sfdxConfigAggregator).to.not.equal(undefined);
+    });
+
+    it('should change back to the original current directory if ConfigAggregator creation fails', async () => {
+      // Arrange
+      configAggregatorCreateSpy.restore();
```

Comment:
interesting...I've never seen this pattern in code before.  Might be simpler to just make it a stub to start or wrap the tests in describe blocks to do the different flavors of setup on the `create` method.

Reply:
moved to a const before the describe